### DISCUSSION TITLE
Green Roofs in UMEP / Solweig

### DISCUSSION THREAD

**Discussion by MaryWP** [UMEP-dev/UMEP#328] [General]
Hello,

we've successfully used UMEP/Solweig for analyzing Outdoor Thermal Comfort.
Is there any experience in using it for projects involving green roofs and/or trees planted on roofs? We are thinking about analyzing Thermal Comfort on these roofs.
For trees, my thirst intuition would be just "planting" them onto the roofs. However, I suspect the output CDSM wouldn't work with the buildings raster. After all, the ground level is different from one tree to the others. 
Green roofs would have two land uses (buildings and grass). 

I would be grateful for any input or experience.

Thank you!

**Comment by biglimp** [comment]
Hi,
Hm, I think there is a way to trick the model without changing any code.
 
If you use the option to derive building footprints from a land cover grid but instead of having a building you specify something else, e.g. grass. Than it might work. I have never tested myself. If you do like that you can also have trees on top of your buildings but beware of the building edges, otherwise part of your trees might end up down in the streets. 
Let me know if you are successful.
/Fredrik

**Comment by rarygit** [comment]
@MaryWP 

My strategy to model the TMRT and UTCI for a rooftop or roof terrace in UMEP is to:

1. Install plugin Serval
https://github.com/lutraconsulting/serval/blob/master/Serval/docs/user_manual.md
2. Select your Building DEM layer in QGS, and select the building/roof/terrace height (raster pixel) using Serval
3. Then select your Ground DEM and change the raster pixel value to the Building Height using Serval (read the Serval manual how to do this).
4. And so forth for all your Building "roof garden" area of interest (AOI).
5. Then change/specify the landcover value for this now elevated Ground DEM to paving or grass etc
6. Select the Building DEM and remove its geometry where it overlies the elevated Ground DEM underneath.

Fairly rudimentary - the key is to match the Ground DEM for the roof garden to at least the respective height of the Building AOI. 
This workaround was accepted by SOLWEIG, but I haven't calibrated or validated this approach.
